features/arithmetic-expressions-in-transformations.md

```diff
@@ -0,0 +1,78 @@
+
+### Supported operators
+| Operator name | Operation |
+| -------- | ----------- |
+| add | add |
+| sub | sub |
+| mul | mul |
+| div | div |
+| mod | mod (remainder)|
+| pow | pow |
+
+
+### Parameter supporting arithmetic expressions
+| Parameter | Examples | Remarks |
+| - | - | - |
+| <p>h</p> |<ul> <li>h-ih_add_10</li> <li>h-ch_mul_0.8_add_bh_mul_0.6</li> </ul> |  |
+| <p>w</p> | <ul> <li>w-ih_add_10</li> <li>w-ch_mul_0.8_add_bh_mul_0.6</li> </ul>|  |
+| <p>ar</p> |<ul> <li>ar-car_mul_0.5</li> <li>ar-bar</li> </ul> | The calculated value of the expression will represent the `width/height` ratio. |
+| <p>x, y</p> | <ul> <li>x-iw_sub_cw</li> <li>y-ih_sub_ch</li> </ul>|  |
+| <p>xc, yc</p> | <ul> <li>xc-cw_mul_0.25</li> <li>yc-ch_mul_0.25</li> </ul>|  |
+| <p>b</p> | <ul> <li>b-ih_mul_0.05_red</li> <li>b-cw_mod_5_AAAA54</li> </ul>| The border follows the `{expression}_{color}` syntax, where the calculated expression value is used as the border width. |
+| <p>lx, ly</p> | <ul> <li>lx-bw_div_2</li> <li>ly-bh_div_2</li> <li>lx-cw_div_2_add_iw_div_2</li>  <li>ly-ch_div_2_add_cw_div_2</li> </ul>| These are layer positional parameters that can be placed in any order within a chain of transformations inside a layer transformation but are always executed last. Hence `ch`, `cw`, and `car` expression variables when used along with `lx` or `ly` parameter always represent the final height, width, and aspect ratio respectively of the asset being overlaid. |
+| <p>pa</p> |<ul> <li>pa-bh_mul_0.05</li> <li>pa-bw_mod_5</li> </ul> | The calculated expression value will be applied as uniform padding along all four edges of the text.  |
+| <p>fs</p> |<ul> <li>fs-bh_mul_0.1</li> <li>fs-bw_div_50</li> </ul> |  |
+
+### Calculation Sequence for multiple operators
+<ol> <li>Multiplication and Division: Multiplication and division are evaluated first from left to right.</li>
+<li>Addition and Subtraction: Addition and subtraction are evaluated next from left to right.</li>
+<li>Modulus: Modulus is evaluated then from left to right.</li>
+<li>Power: Power is evaluated last from left to right.</li></ol>
+
+{% hint style="info" %}
+If you're using expressions inside a layer, then the layer should be a separate chain transformation.
+{% endhint %}
+
+{% hint style="info" %}
+Only the `bw`, `bh`, and `bar` expression variables are allowed inside text layers and solid image layers.
+{% endhint %}
+
+### Examples
+To resize the height to half of the original, the width to one-fourth of the original, and then add a yellow color border that's 5% of the current width, you can use the following URL.
+
+URL - [https://ik.imagekit.io/demo/img/plant.jpeg?tr=w-iw_div_4,h-ih_div_2:b-cw_mul_0.05_yellow](https://ik.imagekit.io/demo/img/plant.jpeg?tr=w-iw_div_4,h-ih_div_2:b-cw_mul_0.05_yellow)
+
+![](https://ik.imagekit.io/demo/img/plant.jpeg?tr=w-iw_div_4,h-ih_div_2:b-cw_mul_0.05_yellow)
+
+
```

Comment:
Why extra new line? It is happening in some other places in this PR as well.

Reply:
addresed